### PR TITLE
move krt registration to after the client is warm

### PR DESCRIPTION
**Pull request by yuval-k**
krt might deadlock if registration happens before client is warm

### CODE REVIEW

**Comment by yuval-k** [comment]
/skip-changlog small bug fix on unreleased version

File: projects/gateway2/proxy_syncer/proxy_syncer.go [hunk 1]
```diff
@@ -449,22 +450,14 @@ func (s *ProxySyncer) Init(ctx context.Context, dbg *krt.DebugHandler) error {
 	s.perclientSnapCollection = snapshotPerClient(logger.Desugar(), dbg, s.uniqueClients, s.mostXdsSnapshots, epPerClient, clustersPerClient)
 
 	// build ProxyList collection as glooProxies change
-	proxiesToReconcile := krt.NewSingleton(func(kctx krt.HandlerContext) *proxyList {
+	s.proxiesToReconcile = krt.NewSingleton(func(kctx krt.HandlerContext) *proxyList {
```
Comment (nfuden) [inline projects/gateway2/proxy_syncer/proxy_syncer.go:453]: we should backlog something to do deadlock detection to see if we can prevent us from writing something like this in the future

File: projects/gateway2/proxy_syncer/proxy_syncer.go [hunk 2]
```diff
@@ -558,6 +544,25 @@ func (s *ProxySyncer) Start(ctx context.Context) error {
...
+	s.statusReport.Register(func(o krt.Event[report]) {
+		if o.Event == controllers.EventDelete {
+			// TODO: handle garbage collection (see: https://github.com/solo-io/solo-projects/issues/7086)
+			return
+		}
+		latestReportQueue.Enqueue(o.Latest().ReportMap)
+	})
+
+	// handler to reconcile ProxyList for in-memory proxy client
+	s.proxiesToReconcile.Register(func(o krt.Event[proxyList]) {
```
[the remark this answers is not in the document]
Comment (yuval-k) [reply projects/gateway2/proxy_syncer/proxy_syncer.go:558]: that's the default in register :)

[the remark this answers is not in the document]
Comment (yuval-k) [reply projects/gateway2/proxy_syncer/proxy_syncer.go:558]: https://github.com/istio/istio/blob/157f73e609e3de7bde9536c69a7f84f3b5891fc2/pkg/kube/krt/collection.go#L581
https://github.com/istio/istio/blob/157f73e609e3de7bde9536c69a7f84f3b5891fc2/pkg/kube/krt/internal.go#L39

**Review by lgadban — APPROVED** [review]

File: projects/gateway2/proxy_syncer/proxy_syncer.go (hunk 1 above)
Comment (yuval-k) [reply projects/gateway2/proxy_syncer/proxy_syncer.go:453]: yes, i'm just not sure what go has to offer here

**Comment by yuval-k** [comment]
/skip-changelog small bug fix on unreleased version

File: projects/gateway2/proxy_syncer/proxy_syncer.go (hunk 1 above)
Comment (lgadban) [reply projects/gateway2/proxy_syncer/proxy_syncer.go:453]: maybe we add it as part of https://github.com/solo-io/solo-projects/issues/7080

**Comment by yuval-k** [comment]
/skip-changelog small bug fix on unreleased version

**Comment by yuval-k** [comment]
/skip-changelog small bug fix on unreleased version

File: projects/gateway2/proxy_syncer/proxy_syncer.go (hunk 1 above)
Comment (nfuden) [reply projects/gateway2/proxy_syncer/proxy_syncer.go:453]: yeah thats actually a really good point.

**Review by nfuden — APPROVED** [review]

**Pull request merged**